drag-and-drop.md

```diff
@@ -0,0 +1,126 @@
+
+*Note: Initiate a drag of a widget from Fyne application to another application is depend on application support the platform (go-gl/glfw) provides. As of now,  the go-gl/glfw changes require to support this. Hence, this topic is moved out of scope for farther discussions as a separate initiative.*
+
+## Architecture / API
+
+Existing interface will not change to preserve compatibility with existing code.
+The proposed changes will extend the interface and give user missing capabilities.
+
+The **fyne.CanvasObject** wishing to initiate drag operation must implement **fyne.Draggable** interface. No interface changes.
+
+The **fyne.CanvasObject** wishing to receiver the drop must implement **fyne.Droppable** interface.
+
+```
+package fyne
+
+type Droppable interface {
+	DragDrop(Draggable)
+}
+```
+
+To receive the drag-and-drop from outside the Fyne application the **fyne.CanvasObject** must implement the **fyne.Droppable** interface.
+
+The **fyne.Draggable** can implement **fyne.DraggableIcon** interface to provide an image that can be used to paint a cursor during drag-and-drop operation.
+
+```
+type DraggableIcon interface {
+	CursorImage() image.Image
+}
+
+```
+
+## Implementation
+
+### Drag-and-Drop within the Fyne application
+
+The drag and drop within the Fyne application is implemented by finding the component under the mouse pointer and supporting **fyne.Droppable** interface in each of the application windows sorted by z-index. Only the first window found with object of **fyne.Droppable** will get the call with fyne.Draggable. This will solve potential issue when there are two or more overlapping windows.
+
+The macOS implementation was tested to confirm this design.
+
+This require for each platform to provide a window position, based on its order from front to back among all visible application windows.
+
+The mobile implementation will be concern only with visible window.
+
+### Drop from outside into Fyne application
+
+The existing go-gl/glfw already provide facilities to receive the drop from another application.
+
+The proposed implementation will utilize this functionality to deliver the drop directly to the object implementing **fyne.Droppable** when found.
+
+
+## Prior art
+
+
+## Drag-and-drop between Fyne app and another application
+
+* start drag operation from Fyne application with a text string to another desktop application (pending external changes)
+
+
+To pass information to receiver the **fyne.Draggable** can also implement **fyne.DragInfo** interface.
+
+```
+type DragInfo interface {
+	Payload() Payload
+}
+```
+
+To initiate a drag-and-drop outside the Fyne application the object implement **fyne.AppDraggable** interface. This interface will allow the object prepare the **fyne.Payload**.
+
+```
+type AppDraggable interface {
```

Comment:
I'm not quite following here... This is for when a drag can start outside the app?
So the payload would be delivered by the toolkit not the object it is landing on.
Also if it is dragging in what is the purpose of `StartDrag`? I would have expected `CanDrop` perhaps?

I wonder if this is complicated by considering fyne and non-fyne apps as different - can we ignore this difference if it implemented well?